[Docs] Add doc for expert parallelism (EP)

YinglingWang opened the pull request:
## Motivation

Add doc for expert parallelism (EP)

## Modifications
1. Update `docs/README.md`: Updated the **"Install Dependency"** section to provide clear installation commands for Debian/Ubuntu, macOS, and CentOS/RHEL environments.
2. Add `docs/advanced_features/expert_parallelism.md`: Following https://github.com/sgl-project/sglang/tree/main/docs to add doc for expert parallelism (EP). We use Markdown instead of Jypyter Notebook because the doc focuses on explaining complex concepts, and executable code for such features would require large MoE models, which are resource-intensive and impractical for standard documentation execution and validation.

## Accuracy Tests

<!-- If this pull request affects model outputs (e.g., changes to the kernel or model forward code), provide accuracy test results. -->

## Benchmarking and Profiling

<!-- If this pull request impacts inference speed, provide benchmarking and profiling results. -->

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

YinglingWang committed:
[Docs] Add doc for expert parallelism

update

zhaochenyang20 commented on docs/README.md:
```diff
@@ -7,8 +7,31 @@ Most documentation files are located under the `docs/` folder.
 
 ### Install Dependency
 
+#### System Dependencies
```
do not use level-4 `####`, instead use `**[content]**`. we only keep level 1~3

zhaochenyang20 committed:
Update README.md

zhaochenyang20 committed:
Refine Expert Parallelism documentation

Updated the formatting and content of the Expert Parallelism documentation, removing redundant information and clarifying key points.

zhaochenyang20 committed:
Refactor Expert Parallelism documentation

Reorganize content for clarity and conciseness in Expert Parallelism documentation. Remove redundant sections and streamline deployment guide.

YinglingWang committed:
address comments

YinglingWang commented on docs/advanced_features/expert_parallelism.md:
```diff
@@ -0,0 +1,65 @@
+
+## Deployment Guide
+
+To enable Expert Parallelism in SGLang, use the relevant flags in your server launch command, which are fully documented in the [Server Arguments page](../advanced_features/server_arguments.md#expert-parallelism). 
```
I am following this https://github.com/sgl-project/sglang/blob/main/docs/README.md#documentation-style-guidelines

> Do not use absolute links (e.g., https://docs.sglang.ai/get_started/install.html). Always prefer relative links (e.g., ../get_started/install.md).

YinglingWang committed:
update command

YinglingWang commented on docs/advanced_features/expert_parallelism.md:
```diff
@@ -0,0 +1,65 @@
+1. MoE models' vast expert weights exceed single GPU memory, and routing tokens across devices incurs complex All-to-All communication overhead. EP partitions expert weights across GPUs to eliminate memory bottlenecks. SGLang integrates [DeepEP](https://github.com/deepseek-ai/DeepEP), a specialized communication library, with Two-Batch Overlap (TBO) to aggressively overlap All-to-All communicatio ...
+2. Real-world inference traffic often leads to uneven expert usage. SGLang's Expert Parallelism Load Balancer ([EPLB](https://github.com/deepseek-ai/eplb)) dynamically monitors expert usage, reassigns experts to balance workload, and maximizes GPU utilization for consistent high performance.
+
+##  Synergy with Prefill-Decode (PD) Disaggregation
```
fixed
